### ISSUE TITLE
Database generated but no data

### ISSUE THREAD

**Issue by bkenobi** [agrif/OctoPrint-InfluxDB#26]
I just installed the plugin on my RPi based OctoPrint setup and configured it to write to a database on a different RPi.  I checked the second RPi and it did generate the DB with all of the sensors I expected.  However, after letting it sit idle for several minutes I do not see any data being added to any of the databases.  It's configured to write every 1 second so I expected the databases to actually become very large very quickly even on idle.  I have not yet tried to print anything so perhaps that is when it starts writing data?  I'll circle back when I print a part later today.

**Comment by bkenobi** [comment]
I printed a part yesterday with the influxdb plugin enabled and no data was stored.  I have the database structure added but it did not populate any data.

Influxdb 1.8.10 on RPi4
Octoprint-influxdb 1.3.2 (current) on RPi3

I can provide other details if I knew what details would help.

![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/74e22ecb-4afa-4d21-8fb5-6d6ca74113be)

**Comment by agrif** [comment]
I'm not quite sure what could be causing this. Are you sure the time on the pi is set correctly? If it's not, that could cause nothing to be returned for the particular data range you are looking at in that query. Check the `date` command on the pi, or consider temporarily removing the time constraints on the query to find out.

If it's not that, could you please attach a copy of your octoprint.log to this issue? You can grab it from "Settings > Octoprint > Logging" on the web UI. I'm confused about what could cause this, but hopefully the log will have *something*.

**Comment by bkenobi** [comment]
I removed the date criteria as recommended and it looks like one data point was generated for the tool0_actual set.

![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/42ecbd03-02aa-43a6-9d4e-2b85657b24ad)

I checked the date command on my octoprint RPI and it shows that it's set correctly (wrong time zone, but it's the right date)
![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/a81627d5-41f2-48ec-901b-f4ef273d4ae3)

I updated my localization info through raspi-config but that shouldn't affect entry into the database.
![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/03bac219-1a28-460c-85c1-4a49af11b7f6)

I

**Comment by bkenobi** [comment]
I should mention that I also checked the date on my influxdb system just to be sure and it was set correctly.  Here is the octoprint log.  I printed a number of parts on 8/15 and 8/16 so there should be plenty in there to look at.

[octoprint(2).log](https://github.com/agrif/OctoPrint-InfluxDB/files/12370740/octoprint.2.log)

And here's the settings screen:
![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/349c1068-a43b-4037-ba10-001f3958f3ae)
![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/1ed028d9-c09f-4eee-ab0b-a6559f234913)

**Comment by agrif** [comment]
The timezone shouldn't matter, as long as the date and time are correct *in that timezone*, which it looks like they are. (If it you want to change it anyway, you can use `raspi-config` and look under localization or internationalization options.)

Oh man, is *that* a suspicious date. Somehow it's generating all the data points at unix time 0.

Can you open a python shell (ideally, the same python used by OctoPrint, which looks like it's `/home/pi/oprint/bin/python`) and run the following two lines?

```
import datetime
print(datetime.datetime.utcnow().isoformat() + 'Z')
```

**Comment by bkenobi** [comment]
That's what I thought when I saw 1969 too!

I launched the version of python in the /home/pi/oprint/bin/python path and it appears to be 2.7.16 even though I thought current Octoprint (1.9.2) should be running python 3.  I ran the commands anyway and it seems to be the right date.

![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/d732ecda-1fb4-4f66-9658-9aee83ea399c)

I also ran the same command in Python3 in case that was the one that was actually being used.

![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/561772a5-d16e-4b4b-8889-6408ac662a31)

**Comment by agrif** [comment]
Your octoprint log shows it running under 3.7.3. To run python from the 'oprint' virtual environment in the directory you were in above, it would be `./python`, as otherwise it just grabs Python from the system. Either way, I don't think this is it.

It looks like my plugin is constantly disconnecting from InfluxDB because it's trying to add a measurement where one field is a different type than existing measurements. I've not seen this before, and I'm not sure which measurement is causing it, but it'd be worth trying to completely delete the octoprint database in Influx and restarting.

**Comment by agrif** [comment]
This actually look exactly like the bug PR #24 is designed to fix -- I need to get off my butt and merge these things.

**Comment by bkenobi** [comment]
I'll delete the octoprint database in influxdb since it's not useful currently anyway.  Maybe it will get it working (who knows).

**Comment by bkenobi** [comment]
I haven't deleted data from influxdb in a while so I'm kinda struggling on format.  I listed all the data in the octoprint database so I figured I'd share that in case it helps prior to deleting.

```
name: temperature
time                bed_actual bed_offset bed_target chamber_offset host   tool0_actual tool0_offset tool0_target
----                ---------- ---------- ---------- -------------- ----   ------------ ------------ ------------
1692301165906093000 27.9       0          0          0              octopi 28.6         0            0
1692301168071284000 27.9       0          0          0              octopi 28.8         0            0
1692301169100901000 27.9       0          0          0              octopi 28.8         0            0
1692301170131918000 27.9       0          0          0              octopi 28.8         0            0
1692301173447272000 27.9       0          0          0              octopi 28.5         0            0
1692301174478933000 27.9       0          0          0              octopi 28.5         0            0
1692301175509506000 27.9       0          0          0              octopi 28.5         0            0
1692301176541256000 27.9       0          0          0              octopi 28.5         0            0
1692301177592944000 27.9       0          0          0              octopi 28.7         0            0
1692301178621827000 27.9       0          0          0              octopi 28.7         0            0
1692301179650050000 27.9       0          0          0              octopi 28.7         0            0
1692301180679810000 27.9       0          0          0              octopi 28.7         0            0
1692301182802021000 28         0          0          0              octopi 28.6         0            0
1692301183833362000 28         0          0          0              octopi 28.6         0            0
1692301184864902000 28         0          0          0              octopi 28.6         0            0
1692301185893721000 28         0          0          0              octopi 28.6         0            0
1692301186915704000 28         0          0          0              octopi 28.6         0            0
1692301187953612000 28.1       0          0          0              octopi 28.6         0            0
1692301188976827000 28.1       0          0          0              octopi 28.6         0            0
1692301190006559000 28.1       0          0          0              octopi 28.6         0            0
1692301191031739000 28.1       0          0          0              octopi 28.6         0            0
1692301192061933000 28         0          0          0              octopi 28.7         0            0
1692301193094634000 28         0          0          0              octopi 28.7         0            0
1692301194123484000 28         0          0          0              octopi 28.7         0            0
1692301195151611000 28         0          0          0              octopi 28.7         0            0
1692301196180042000 28         0          0          0              octopi 28.7         0            0
1692301197206957000 28         0          0          0              octopi 28.6         0            0
1692301198237322000 28         0          0          0              octopi 28.6         0            0
1692301199268636000 28         0          0          0              octopi 28.6         0            0
1692301200299591000 28         0          0          0              octopi 28.6         0            0
1692301201332041000 28         0          0          0              octopi 28.6         0            0
1692301202505956000 28         0          0          0              octopi 28.6         0            0
1692301203535332000 28         0          0          0              octopi 28.6         0            0
1692301204562317000 28         0          0          0              octopi 28.6         0            0
1692301207002889000 27.9       0          0          0              octopi 28.7         0            0
1692301208031840000 27.9       0          0          0              octopi 28.7         0            0
1692301209780905000 27.9       0          0          0              octopi 28.7         0            0
1692301210811756000 27.9       0          0          0              octopi 28.7         0            0
1692301212536561000 28         0          0          0              octopi 28.5         0            0
1692301213564545000 28         0          0          0              octopi 28.5         0            0
1692301215115558000 28         0          0          0              octopi 28.5         0            0
1692301216147763000 28         0          0          0              octopi 28.5         0            0
1692301218315634000 27.8       0          0          0              octopi 28.7         0            0
1692301219345719000 27.8       0          0          0              octopi 28.7         0            0
1692301220898521000 27.8       0          0          0              octopi 28.7         0            0
1692301221980157000 28.1       0          0          0              octopi 28.6         0            0
1692301223014203000 28.1       0          0          0              octopi 28.6         0            0
1692301226311494000 28.1       0          0          0              octopi 28.6         0            0
1692301228249250000 27.9       0          0          0              octopi 28.6         0            0
1692301229305664000 27.9       0          0          0              octopi 28.6         0            0
1692301230337093000 27.9       0          0          0              octopi 28.6         0            0
1692301231365430000 27.9       0          0          0              octopi 28.6         0            0
1692301232543809000 28         0          0          0              octopi 28.6         0            0
1692301234014837000 28         0          0          0              octopi 28.6         0            0
1692301235880589000 28         0          0          0              octopi 28.6         0            0
1692301237760160000 27.9       0          0          0              octopi 28.6         0            0
1692301239306395000 27.9       0          0          0              octopi 28.6         0            0
1692301240856585000 27.9       0          0          0              octopi 28.6         0            0
1692301241887562000 27.9       0          0          0              octopi 28.6         0            0
1692301242918880000 28         0          0          0              octopi 28.7         0            0
```

**Comment by bkenobi** [comment]
Actually now that I look at it, whatever I did when trying to clear the data from influxdb seems to have woken it up.  I seem to have data from right now in the database.

![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/94252858-56e7-4085-a5ff-50f8ec4ac9f2)

**Comment by agrif** [comment]
Strange, but I'll take it. I'll try to get that PR merged (tomorrow, hopefully) and cut a new release and maybe prevent this from happening again.

**Comment by bkenobi** [comment]
Out of curiousity, is it possible for you to set the retention policy when you create the database?  If the data is recorded at 1s intervals and the printer is left on after each print, the database will get filled with useless info.  I was thinking either having a retention policy that only keeps data for 1 month or maybe a second set of controls in octoprint that keeps print output at 1s (or whatever) and a second control that stores non-print temps either at a lower frequency or not at all.

**Comment by bkenobi** [comment]
And it's definitely working now.
![image](https://github.com/agrif/OctoPrint-InfluxDB/assets/6903751/886584d3-4259-4d81-a04d-711b4fd6c1ff)

**Comment by agrif** [comment]
I've thought about setting a retention policy, but there's so many knobs to touch there that it seems outside the scope of the plugin. Creating the database automatically is easy and unlikely to do anything *that* surprising, but messing with retention policies might delete something it shouldn't if I'm not careful.

**Comment by bkenobi** [comment]
I agree about the retention policy.  It's a bit scary to think you could break someone's data outside the scope of the plugin!

**Comment by agrif** [comment]
I've merged that PR into version 1.4.0, available now, and I've made a new issue for the slower interval idea. I'm closing this issue, feel free to open another if you run into any more issues!
